dali/base/dadfs.cpp

```diff
@@ -8239,20 +8239,19 @@ class CNamedGroupStore: implements INamedGroupStore, public CInterface
 
 };
 
-static CNamedGroupStore *groupStore = NULL;
-static CriticalSection groupsect;
+static std::atomic<CNamedGroupStore *> groupStore{nullptr};
 
 bool CNamedGroupIterator::match()
 {
     if (conn.get()) {
         if (matchgroup.get()) {
-            if (!groupStore)
+            if (!groupStore.load())
                 return false;
             const char *name = pe->query().queryProp("@name");
             if (!name||!*name)
                 return false;
             GroupType dummy;
-            Owned<IGroup> lgrp = groupStore->dolookup(name, conn, NULL, dummy);
+            Owned<IGroup> lgrp = groupStore.load()->dolookup(name, conn, NULL, dummy);
```

Comment:
@jakesmith theoretically this could crash if closedownDFS is called between the test and this call.  I assume that is not worth worrying about.

Reply:
not worth worrying about for this PR/issue, and not worth seriously worrying about in general I think.
I think it's fine as is.